ethereum/contracts/Messages.sol

```diff
@@ -80,6 +80,14 @@ contract Messages is Getters {
             require(i == 0 || sig.guardianIndex > lastIndex, "signature indices must be ascending");
             lastIndex = sig.guardianIndex;
 
+            /// @dev Ensure that the provided signature index is within the
+            /// bounds of the guardianSet. This is implicitly checked by the array
+            /// index operation below, so this check is technically redundant.
+            /// However, reverting explicitly here ensures that a bug is not
+            /// introduced accidentally later due to the nontrivial storage
+            /// semantics of solidity.
+            require(sig.guardianIndex < guardianSet.keys.length, "guardian index out of bounds");
```

Comment:
Should we store guardianSet.keys.length as a local variable instead of accessing .length in each loop?

Reply:
yes 👍